synnax/pkg/distribution/channel/lease_proxy.go

```diff
@@ -137,17 +142,65 @@ func (lp *leaseProxy) createFreeVirtual(
 	if lp.freeCounter == nil {
 		panic("[leaseProxy] - tried to assign virtual keys on non-bootstrapper")
 	}
-	toCreate, err := lp.maybeRetrieveExisting(ctx, tx, channels, lp.freeCounter, retrieveIfNameExists)
+
+	// If existing channels are passed in, update as necessary (for calc channels)
+	// unless retrieveIfNameExists is true,
```

Comment:
Formatting in this comment is strange. Also, the last sentence should say "if retrieveIfNameExists is true and the user has provided channels to update, we need to re-set those channels to their actual values to ensure the user does not mistakenly think the update was successfuf"